qiskit/visualization/matplotlib.py

```diff
@@ -596,6 +588,11 @@ def _get_text_width(self, text, fontsize, param=False):
         # If there are subscripts or superscripts in mathtext string
         # we need to account for that spacing by manually removing
         # from text string for text length
+
+        # if it's a register and there's a subscript at the end,
+        # remove 1 underscore, otherwise don't remove any
+        if reg is not None:
+            num_underscores = reg
```

Comment:
I think this comment might be out of date now?  Either that, or I'm not sure what the `reg` parameter is meant to be.

Reply:
I changed the name in https://github.com/Qiskit/qiskit-terra/pull/7459/commits/1fa466eb5b68a6c00ed2dd5e7c1669b097ac6a0d to `reg_to_remove`, though open to a better name. It's how many underscores to remove 1 or 0 or if None, all of them.

The whole `_get_text_width` function probably needs to be rethought a bit. The `latex_to_text `was added to solve a particular problem, but I don't think it works correctly for widths other than registers. Probably for another issue.